File: scripts/path_config.py
from __future__ import annotations

import os
from dataclasses import dataclass
from pathlib import Path


@dataclass(frozen=True)
class RepositoryPaths:
    repo_root: Path
    external_data_root: Path
    work_root: Path
    data_public_root: Path
    results_root: Path


def get_paths() -> RepositoryPaths:
    repo_root = Path(__file__).resolve().parents[1]
    override = os.environ.get("GART_UC_DATA_ROOT", "").strip()
    external = Path(override).expanduser().resolve() if override else repo_root / "data_external"
    return RepositoryPaths(repo_root, external, repo_root / "work", repo_root / "data_public", repo_root / "results_reproduced")


def require_file(path: Path, label: str = "required input") -> Path:
    path = Path(path)
    if not path.is_file():
        raise FileNotFoundError(f"Required external input not found: {path} ({label}). See data_external/README.md or set GART_UC_DATA_ROOT.")
    return path

File: scripts/path_config.R
gart_repo_root <- function(script_file) {
  override <- Sys.getenv("GART_UC_REPO_ROOT", unset = "")
  if (nzchar(override)) return(normalizePath(override, winslash = "/", mustWork = FALSE))
  normalizePath(file.path(dirname(script_file), "..", ".."), winslash = "/", mustWork = FALSE)
}

gart_paths <- function(script_file) {
  repo <- gart_repo_root(script_file)
  override <- Sys.getenv("GART_UC_DATA_ROOT", unset = "")
  external <- if (nzchar(override)) normalizePath(override, winslash = "/", mustWork = FALSE) else file.path(repo, "data_external")
  list(repo_root = repo, external_data_root = external, work_root = file.path(repo, "work"), data_public_root = file.path(repo, "data_public"), results_root = file.path(repo, "results_reproduced"))
}

gart_require_file <- function(path, label = "required input") {
  if (!file.exists(path)) stop(sprintf("Required external input not found: %s (%s). See data_external/README.md or set GART_UC_DATA_ROOT.", path, label), call. = FALSE)
  path
}
